StoreMessageContentActivity/handler.ts

```diff
@@ -53,13 +67,107 @@ export type StoreMessageContentActivityResult = t.TypeOf<
+  kind: t.keyof({ ERROR: null, LEGACY: null }),
+  message: t.string
+});
+
+export type ServicePreferenceError = t.TypeOf<typeof ServicePreferenceError>;
+
+export type ServicePreferenceValueOrError = (params: {
+  readonly serviceId: NonEmptyString;
+  readonly fiscalCode: FiscalCode;
+  readonly userServicePreferencesMode: ServicesPreferencesMode;
+  readonly userServicePreferencesVersion: number;
+}) => TaskEither<
+  ServicePreferenceError,
+  ReadonlyArray<BlockedInboxOrChannelEnum>
+>;
+
+type ServicePreferencesValues = Omit<
+  ServicePreference,
+  "serviceId" | "fiscalCode" | "settingsVersion"
+>;
+
+const channelToBlockedInboxOrChannelEnum: {
+  readonly [key in keyof ServicePreferencesValues]: BlockedInboxOrChannelEnum;
+} = {
+  isEmailEnabled: BlockedInboxOrChannelEnum.EMAIL,
+  isInboxEnabled: BlockedInboxOrChannelEnum.INBOX,
+  isWebhookEnabled: BlockedInboxOrChannelEnum.WEBHOOK
+};
+
+const servicePreferenceToBlockedInboxOrChannels: (
+  servicePreference: ServicePreference
+) => ReadonlyArray<BlockedInboxOrChannelEnum> = servicePreference =>
+  Object.entries(servicePreference)
+    // take only attributes of ServicePreferencesValues
+    .filter(([name, _]) => channelToBlockedInboxOrChannelEnum[name])
+    // take values set to false
+    .filter(([_, isEnabled]) => !isEnabled)
+    // map to BlockedInboxOrChannelEnum
+    .map(([name, _]) => channelToBlockedInboxOrChannelEnum[name]);
+
+const getServicePreferenceValueOrError = (
+  servicePreferencesModel: ServicesPreferencesModel
+): ServicePreferenceValueOrError => ({
+  fiscalCode,
+  serviceId,
+  userServicePreferencesMode,
+  userServicePreferencesVersion
+}): TaskEither<
+  ServicePreferenceError,
+  ReadonlyArray<BlockedInboxOrChannelEnum>
+> =>
+  taskEither
+    .of<ServicePreferenceError, ServicesPreferencesMode>(
+      userServicePreferencesMode
+    )
+    .chain(
+      fromPredicate(
+        _ => _ !== ServicesPreferencesModeEnum.LEGACY,
+        () => ({
+          kind: "LEGACY",
+          message: "User service preferences mode is LEGACY"
+        })
+      )
+    )
+    .map(() =>
+      makeServicesPreferencesDocumentId(
+        fiscalCode,
+        serviceId,
+        userServicePreferencesVersion as NonNegativeInteger
+      )
+    )
+    .chain(documentId =>
+      servicePreferencesModel
+        .find([documentId, fiscalCode])
+        .mapLeft<ServicePreferenceError>(failure => ({
+          kind: "ERROR",
+          message: `COSMOSDB|ERROR=${failure.kind}`
+        }))
+    )
+    .map(maybeServicePref =>
+      maybeServicePref.foldL<ReadonlyArray<BlockedInboxOrChannelEnum>>(
+        () =>
+          // if we do not have a preference we return an empty array only
+          // if we have preference mode AUTO, else we must return an array
+          // with BlockedInboxOrChannelEnum.INBOX
+          userServicePreferencesMode === ServicesPreferencesModeEnum.AUTO
+            ? []
+            : [BlockedInboxOrChannelEnum.INBOX],
```

Comment:
This is ok so far, but what if we add another way to handle subscription for our users (that is: we add another case to `ServicesPreferencesModeEnum`)? Such new case would be considered in the _else_ case, just like `MANUAL`, and we wouldn't notice that.

My advice is to always be explicit when handling enums, and treat unexpected values for what they are: values for which we cannot make decisions.

[Here](https://github.com/pagopa/io-functions-services/blob/2df422b7ba2e582f0a4b8e4030a0b6a854b3c154/utils/profile.ts#L116) you can see an example

Reply:
I check.

Reply:
Resolved in https://github.com/pagopa/io-functions-services/pull/142/commits/b50a4f98a99bbfe99da4326619f7b1434660e716